pywr/solvers/cython_glpk.pyx

```diff
@@ -211,8 +218,27 @@ cdef class CythonGLPKSolver:
             free(ind)
             free(val)
 
+        # virtual storage
+        if len(virtual_storages):
+            self.idx_row_virtual_storages = glp_add_rows(self.prob, len(virtual_storages))
+        for col, storage in enumerate(virtual_storages):
+            # We need to handle the same route appearing twice here.
+            cols = [n for n, route in enumerate(routes) for some_node in route if some_node in storage.nodes]
+            # Use Counter to find the number of unique entries for routes
+            cntr = Counter(cols)
+
+            ind = <int*>malloc((1+len(cntr)) * sizeof(int))
+            val = <double*>malloc((1+len(cntr)) * sizeof(double))
+            for n, (c, num) in enumerate(cntr.items()):
+                ind[1+n] = self.idx_col_routes+c
+                val[1+n] = -num
+
+            glp_set_mat_row(self.prob, self.idx_row_virtual_storages+col, len(cntr), ind, val)
+            free(ind)
+            free(val)
+
         # aggregated node flow ratio constraints
-        self.idx_row_aggregated = self.idx_row_storages + len(storages)
+        self.idx_row_aggregated = self.idx_row_storages + len(virtual_storages)
```

Comment:
This should be:

```
self.idx_row_aggregated = self.idx_row_virtual_storages + len(virtual_storages)
```

Alternatively, we could set `self.idx_row_aggregated = None`, then in the loop when the first row is created (`if ... is None`) assign it the value returned by `glp_add_rows`.